catalyst_voices/packages/internal/catalyst_voices_shared/lib/src/extension/map_to_list_ext.dart

```diff
@@ -0,0 +1,28 @@
+extension MapToListExt on Map<String, dynamic> {
+  List<Map<String, dynamic>> convertMapToListWithIds() {
+    final list = <Map<String, dynamic>>[];
+
+    for (final entry in entries) {
+      if (entry.key == r'$schema') continue;
+      final value = entry.value as Map<String, dynamic>;
+      value['id'] = entry.key;
+      list.add(value);
+    }
+
+    return list;
+  }
+
+  List<Map<String, dynamic>> convertMapToListWithIdsAndValues() {
+    final list = <Map<String, dynamic>>[];
+
+    for (final entry in entries) {
+      if (entry.key == r'$schema') continue;
```

Comment:
You are hardcoding here logic which is relevant only to schemas and documents, this can't be a generic map extension. I would put it under `lib/src/document/extension/map_ext.dart` and call it `DocumentMapToListExt` to mark it as only compatible with documents/schemas.